Installation/nnAudio/features/vqt.py

```diff
@@ -0,0 +1,202 @@
+            if verbose == True:
+                print("Creating early downsampling filter ...", end='\r')
+            start = time()
+            sr, self.hop_length, self.downsample_factor, early_downsample_filter, \
+                self.earlydownsample = get_early_downsample_params(sr,
+                                                                   hop_length,
+                                                                   fmax_t,
+                                                                   Q,
+                                                                   self.n_octaves,
+                                                                   verbose)
+            self.register_buffer('early_downsample_filter', early_downsample_filter)
+            
+            if verbose==True:
+                print("Early downsampling filter created, \
+                        time used = {:.4f} seconds".format(time()-start))
+        else:
+            self.downsample_factor = 1.
+        
+        # For normalization in the end
+        # The freqs returned by create_cqt_kernels cannot be used
+        # Since that returns only the top octave bins
+        # We need the information for all freq bin
+        alpha = 2.0 ** (1.0 / bins_per_octave) - 1.0
+        freqs = fmin * 2.0 ** (np.r_[0:n_bins] / np.float(bins_per_octave))
+        self.frequencies = freqs
+        lenghts = np.ceil(Q * sr / (freqs + gamma / alpha))
+        
+        # get max window length depending on gamma value
+        max_len = int(max(lenghts))
+        self.n_fft = int(2 ** (np.ceil(np.log2(max_len))))
+
+        lenghts = torch.tensor(lenghts).float()
+        self.register_buffer('lenghts', lenghts)
+            
+
+    def forward(self, x, output_format=None, normalization_type='librosa'):
+        """
+        Convert a batch of waveforms to VQT spectrograms.
+        
+        Parameters
+        ----------
+        x : torch tensor
+            Input signal should be in either of the following shapes.\n
+            1. ``(len_audio)``\n
+            2. ``(num_audio, len_audio)``\n
+            3. ``(num_audio, 1, len_audio)``
+            It will be automatically broadcast to the right shape
+        """              
+        output_format = output_format or self.output_format
+        
+        x = broadcast_dim(x)
+        if self.earlydownsample==True:
+            x = downsampling_by_n(x, self.early_downsample_filter, self.downsample_factor)
+        hop = self.hop_length
+        vqt = []
+
+        x_down = x  # Preparing a new variable for downsampling
+        my_sr = self.sr
+
+        for i in range(self.n_octaves):
+            if i > 0:
+                x_down = downsampling_by_2(x_down, self.lowpass_filter)
+                my_sr /= 2
+                hop //= 2
+
+            else:
+                x_down = x
+            
+            Q = float(self.filter_scale)/(2**(1/self.bins_per_octave)-1)
```

Comment:
Sorry for replying late. You can understand this Q as an "initial value of Q".
Following librosa's implementation, the "variable" part actually lies in `lengths` of `create_ cqt_kernels`. The `gamma` will change the window lengths as compared to CQT. Since window length and Q are interrelated, by changing the window lengths we can also view it as "variable-Q".

![image](https://user-images.githubusercontent.com/13913915/146647970-9639c471-f5da-497a-8872-3e19aa1a4ed8.png)